### PR TITLE
[Simplified Login] Prologue screen changes

### LINKED ISSUE #7610: [Simplified login] Prologue Screen changes
- Update image
- Reword the “new to WooCommerce” button
- Reword the Email login button.
- Replace the "login with site address" button with the store creation flow (To be confirmed)

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #7610 

### Description

This PR implements the following Prologue screen designs (these are based on the iOS version):

| simplified login variant | control variant |
|-|-|
|  |  |

The PR includes various small design updates, such as:
1. Making the design work on smaller screen (tested with a 4" screen)
2. updates to the sw600dp and landscape layouts
3. the "OR" divider on control layout, above the "Get started" button

### Testing instructions

Run app in logged out state and compare the shown prologue screen with the design above.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/layout/fragment_login_prologue.xml [hunk 1]
```diff
@@ -69,45 +69,61 @@
...
-        app:layout_constraintBottom_toTopOf="@id/button_login_wpcom"
+        app:layout_constraintBottom_toTopOf="@id/loginButtons"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/prologueTitle" />
 
-    <com.google.android.material.button.MaterialButton
-        android:id="@+id/button_login_wpcom"
-        style="@style/Woo.Button.Colored"
+    <LinearLayout
```
Comment (hafizrahman) [inline WooCommerce/src/main/res/layout/fragment_login_prologue.xml:66]: The initial layout was a bit complex to my brain with the differing buttons's placements / constraint positioning. 

With how some buttons are now shown and hidden depending on the variant, I thought it'll be easier to follow when everything is put in a LinearLayout.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt [hunk 2]
```diff
@@ -60,6 +66,16 @@ open class LoginPrologueFragment(@LayoutRes layout: Int) : Fragment(layout) {
...
         }
+
+        // Various updates related to Simplified Login A/B testing.
+        if (isSimplifiedLoginVariant) {
+            binding.newToWooButton.text = getString(R.string.login_prologue_learn_more_about_woo)
+
+            binding.buttonLoginStore.hide()
+            binding.buttonGetStarted.hide()
+
+            binding.buttonCreateStore.isVisible = FeatureFlag.STORE_CREATION_WEBVIEW_FLOW.isEnabled()
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt]: Button doesn't do anything for now, let's make separate PR to make it open to the right flow as I haven't checked the status / availability for it.

File: WooCommerce/src/main/res/layout/fragment_login_prologue.xml [hunk 3]
```diff
@@ -59,55 +59,90 @@
...
+            android:id="@+id/button_get_started"
+            style="@style/Woo.Button.Colored.White"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/major_100"
+            android:layout_marginEnd="@dimen/major_100"
+            android:layout_marginBottom="@dimen/major_150"
+            android:text="@string/signup_get_started_button"
+            android:textAllCaps="false"
+            android:visibility="gone" />
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/layout/fragment_login_prologue.xml]: I think we should have just a single button here, whether it's called "Get Started" or "Create a store" is TBD (check the comment here ERGtCFLyiC9CFpHsw1XIid-fi-411%3A6226), but it's the same button with the same behavior.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt]: As I explained below, the "Create Store" and "Get Started" should be a single button, and it depends on `FeatureFlag.ACCOUNT_CREATION_FLOW`.

**Comment by hafizrahman** [comment]
This is ready for another review @hichamboushaba :)

File: WooCommerce/src/main/res/layout/fragment_login_prologue.xml (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/res/layout/fragment_login_prologue.xml:66]: Yes, when possible, a LinearLayout is always better and more performant than a ConstraintLayout.

The initial layout was actually broken 😄, it caused some issues like this p1666951087613899-slack-C70PAM3RA, so thanks for updating it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt [hunk 4]
```diff
@@ -94,7 +94,10 @@ class FirebaseRemoteConfigRepository @Inject constructor(
             }
 
     override fun getSimplifiedLoginVariant(): LoginVariant {
-        return try {
+        return if (PackageUtils.isTesting()) {
+            // Use control variant to prevent e2e test breakages
+            LoginVariant.valueOf(defaultValues[SIMPLIFIED_LOGIN_VARIANT_KEY]!!)
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt]: You are returning the simplified login variant here, I assume it's just an error, you should return the Control variant.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt [hunk 5]
```diff
@@ -32,9 +36,13 @@ open class LoginPrologueFragment(@LayoutRes layout: Int) : Fragment(layout) {
     constructor() : this(R.layout.fragment_login_prologue)
 
     @Inject lateinit var unifiedLoginTracker: UnifiedLoginTracker
+    @Inject lateinit var simplifiedLoginExperiment: SimplifiedLoginExperiment
     private var prologueFinishedListener: PrologueFinishedListener? = null
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        simplifiedLoginExperiment.activate()
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt]: I think it's better to check the `savedInstanceState` value before triggering `activate`, just to make sure it's triggered only once.

File: WooCommerce/src/main/res/layout/fragment_login_prologue.xml (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/res/layout/fragment_login_prologue.xml:66]: When I replied above, I assumed you changed the root layout to LinearLayout too.
I think since you moved the buttons to their own layout, it's better to change the root layout too, it'll make things clearer and easier to follow, and it'll make handling the resizing of the image easier when using weights. just a suggestion 🙂.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt (hunk 4 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt]: It's returning `defaultValues`, though, which should return Control 🤔

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt]: Ah, sorry, you are right, I didn't read the code correctly here.

But still, just a np, I would use the `LoginVariant` directly here, instead of passing by the `defaultValue`, just in case we wanted to change the defaultValue in the future, it would be best to keep the UI tests value hardcoded to the correct value.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt (hunk 5 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt]: We discussed this and decided it's OK for it to be run multiple times.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt (hunk 4 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt]: Good point!

File: WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml [hunk 6]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (hafizrahman) [inline WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml:1]: Since we don't have actual design for this, I'm basically just updating this based on what I feel would be a good enough design.

File: WooCommerce/src/main/res/layout-sw600dp/fragment_login_prologue.xml [hunk 7]
```diff
@@ -35,32 +35,35 @@
 
     <ImageView
         android:id="@+id/imageView"
```
Comment (hafizrahman) [inline WooCommerce/src/main/res/layout-sw600dp/fragment_login_prologue.xml:37]: Same as landscape layout: Since we don't have actual design for this, I'm basically just updating this based on what I feel would be a good enough design.

File: WooCommerce/src/main/res/layout/fragment_login_prologue.xml [hunk 8]
```diff
@@ -9,105 +9,150 @@
...
-        android:layout_marginBottom="@dimen/major_150"
+        android:layout_marginTop="@dimen/minor_100"
         android:text="@string/login_prologue_new_to_woo"
         android:textAllCaps="false"
-        app:layout_constraintBottom_toTopOf="@id/button_login_wpcom"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/prologueTitle" />
+        app:layout_constraintTop_toBottomOf="@id/prologueTitle"
+        android:visibility="gone" />
```
Comment (hafizrahman) [inline WooCommerce/src/main/res/layout/fragment_login_prologue.xml]: `newToWooButton` is now supposed to be hidden both on control and prologue, but I'm choosing not to delete it outright yet just in case it needs to be re-added later.

Comment (hichamboushaba) [reply WooCommerce/src/main/res/layout/fragment_login_prologue.xml]: Personally, I think we should remove it, a good read on why https://kentcdodds.com/blog/please-dont-commit-commented-out-code

Comment (hafizrahman) [reply WooCommerce/src/main/res/layout/fragment_login_prologue.xml]: Good point. I removed it.

File: WooCommerce/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -204,6 +204,7 @@
     <string name="login_with_store_address">Log in with your store address</string>
     <string name="login_configure_link">Read the %1$sconfiguration instructions%2$s.</string>
     <string name="login_with_qr_code">Scan QR code to login</string>
+    <string name="login_create_store">Create a Store</string>
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/values/strings.xml]: I don't think this is used anywhere?

**Review by hichamboushaba — APPROVED** [review]
Nice job @hafizrahman 💯, I left one comment about one unused string resource, otherwise it's good to go.

File: WooCommerce/src/main/res/values/strings.xml (hunk 9 above)
Comment (hafizrahman) [reply WooCommerce/src/main/res/values/strings.xml]: Thanks!
